### ISSUE TITLE
[Question] About the reconnection

### ISSUE THREAD

**Issue by thehaung** [sijms/go-ora#333]
Hi @sijms and folks,

I wondering, when my services in runtime, in particular the OracleDB restart, how can I reconnect to the DB. Do you have an examples or something else related to my question?

**Comment by sijms** [comment]
use urlOptions 
```golang
urlOptions := map[string]string{
	"FAILOVER": "5", // the number represent how many time driver will try to reconnect before fail
	"TRACE FILE": "trace.log",
}
```

**Comment by thehaung** [comment]
This will be perform when I inject the Oracle connection into other package in my service right?

**Comment by thehaung** [comment]
Sample code I try with your provide options
```
func (s staffRepository) List(ctx context.Context) ([]domain.Staff, error) {
	rows, err := s.oracleDB.Query("SELECT ID, NAME, TEAM_NAME, ORGANIZATION, TITLE, ONBOARD_DATE, ACTIVE, CREATED_AT, MODIFIED_AT FROM " +
		_staffTableName + " WHERE ACTIVE = 1 ORDER BY CREATED_AT DESC")
	if err != nil {
		fmt.Println("rows query: ", err)
		return nil, err
	}
	defer func() {
		err = rows.Close()
		if err != nil {
			fmt.Println("Can't close dataset: ", err)
		}
	}()
}
```

Error when I restart the DB for reproduce my situation
```
rows query:  EOF
rows query:  ORA-06413: Connection not open
rows query:  ORA-06413: Connection not open
rows query:  ORA-06413: Connection not open
rows query:  ORA-06413: Connection not open
```

**Comment by sijms** [comment]
I am working on it

**Comment by sijms** [comment]
I fix it in v2.6.9
what I make as follow
check for 2 errors io.EOF and syscall.EPIPE before the following operation
1- query  ==> reconnect and then requery
2- fetch. ==> reconnect and return ora-03135
3- exec. ==>  reconnect and return ora-03135
4- ping. ==>  reconnect and return ora-03135
5- commit ==> reconnect and return ora-03135
6- rollback ==>  reconnect and return ora-03135
7- ref_cursor query ==> reconnect and then requery

would you please test and give me your feedback

**Comment by thehaung** [comment]
Oke @sijms, I will do this in a next day

**Comment by sijms** [comment]
ok no problem

**Comment by sijms** [comment]
would you share trace.log file

**Comment by sijms** [comment]
log file should contain a line related to this
```golang
tracer.Print("reconnect trial #", trial)
```

**Comment by sijms** [comment]
also confirm that you don't close the connection when you receive an error

**Comment by thehaung** [comment]
> also confirm that you don't close the connection when you receive an error

Sure

**Comment by thehaung** [comment]
> would you share trace.log file

https://github.com/thehaung/golang-oracle-example/blob/master/trace.log

Here is my log file and my repo I am working on @sijms

**Comment by sijms** [comment]
i see the log file
you reach up to [10 reconnect trial](https://github.com/thehaung/golang-oracle-example/blob/5d3a8d52d0dbb3ae45ca56f59e3d2cba2b65128e/trace.log#L1235)
after that [return connection refuse](https://github.com/thehaung/golang-oracle-example/blob/5d3a8d52d0dbb3ae45ca56f59e3d2cba2b65128e/trace.log#L1268)

**Comment by sijms** [comment]
know i understand your issue
you need to add new url option `"RETRYTIME": 10` so there will be a sufficient time for oracle server to restarted

**Comment by thehaung** [comment]
```
"RETRYTIME": 10
``` 
The 10 is ms or what? I trying but this does not work!

**Comment by sijms** [comment]
still not added yet I will add it and it will represent the time between each reconnection trial so server has enough time to startup

**Comment by sijms** [comment]
fixed in v2.6.11

**Comment by thehaung** [comment]
Oke I will try this, in a next day,  If this really fixed, I'll create PR for this Example
